### PR TITLE
[#1501] Create Unit Tests for FeedbackConstantSumQuestionDetails class

### PR DESCRIPTION
**Pull request by andremralves**
Fixes part of #1501

Added unit tests for shouldChangesRequireResponseDeletion method of FeedbackConstantSumQuestionDetails class. It was used the multiple-condition technique to create the test cases and increase the class coverage.

### CODE REVIEW

File: src/test/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetailsTest.java [hunk 1]
```diff
@@ -151,6 +151,108 @@ public void testValidateResponseDetails_amongOptionsInvalidAnswer_shouldReturnNo
...
+    @Test
+    public void testShouldChangesRequireResponseDeletion_differentConstSumOptions_shouldReturnTrue() {
+        FeedbackConstantSumQuestionDetails feedbackQuestionDetails = new FeedbackConstantSumQuestionDetails();
+        feedbackQuestionDetails.setConstSumOptions(Arrays.asList("a", "b"));
+
+        FeedbackConstantSumQuestionDetails newFeedbackQuestionDetails = new FeedbackConstantSumQuestionDetails();
+        newFeedbackQuestionDetails.setConstSumOptions(Arrays.asList("a", "b", "c"));
+
+        assertTrue(feedbackQuestionDetails.shouldChangesRequireResponseDeletion(newFeedbackQuestionDetails));
+        assertTrue(newFeedbackQuestionDetails.shouldChangesRequireResponseDeletion(feedbackQuestionDetails));
```
Comment (zhaojj2209) [inline src/test/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetailsTest.java:163]: This is the only test case with two `assertTrue`s, I would suggest removing the extra `assertTrue` to make it consistent with the rest of the `testShouldChangesRequireResponseDeletion` tests.

**Review by zhaojj2209 — COMMENTED** [review]
Thank you for adding the additional tests!

If you wish to, you can also consider adding more tests of the `testShouldChangesRequireResponseDeletion_sameXXX_shouldReturnFalse()` variety.

File: src/test/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetailsTest.java (hunk 1 above)
Comment (andremralves) [reply src/test/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetailsTest.java:163]: Thank you for the feedback! The idea behind of this test case was to test both conditions of the first if statement, one with the first list being bigger than the second and vice versa. If I remove one of these, the jacoco's report will show a yellow line meaning that just one of the conditions are being covered.

Comment (zhaojj2209) [reply src/test/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetailsTest.java:163]: > Thank you for the feedback! The idea behind of this test case was to test both conditions of the first if statement, one with the first list being bigger than the second and vice versa. If I remove one of those, the jacoco's report will show a yellow line meaning that just one of the conditions is being covered. Maybe I can split this test case in two others, each testing one of the conditions. What do you think about it?

Thanks for the clarification! There's no further changes needed in that case.

**Review by wkurniawan07 — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**